### PR TITLE
deb: tweak lintian warnings

### PR DESCRIPTION
**Pull request by kenhys**
* Drop unknown tag
   * Ignore E: unstripped-binary-or-object
    * Fix W: transitional-package-not-oldlibs-optional
    * Suppress W: unusual-interpreter
    * Suppress W: debian-rules-should-not-use-custom-compression-settings
    * Suppress E: custom-library-search-path
    * Suppress W: custom-compression-in-debian-rules

### CODE REVIEW

File: fluent-package/debian/fluent-package.lintian-overrides [hunk 1]
```diff
@@ -31,3 +26,5 @@ fluent-package: binary-or-shlib-defines-rpath opt/fluent/lib/ruby/gems/*/gems/no
 fluent-package: embedded-library opt/fluent/lib/ruby/gems/*/extensions/x86_64-linux/*/nokogiri-*/nokogiri/nokogiri.so: libxml2
 fluent-package: embedded-library opt/fluent/lib/ruby/gems/*/gems/nokogiri-*/ext/nokogiri/nokogiri.so: libxml2
 fluent-package: embedded-library opt/fluent/lib/ruby/gems/*/gems/nokogiri-*/lib/nokogiri/nokogiri.so: libxml2
+fluent-package: unstripped-binary-or-object opt/fluent/lib/ruby/gems/3.2.0/extensions/x86_64-linux/3.2.0/debug-1.7.1/debug/debug.so
+fluent-package: unusual-interpreter opt/fluent/lib/ruby/gems/3.2.0/gems/ffi-1.15.5/ext/ffi_c/libffi/generate-darwin-source-and-headers.py #!python
```
Comment (cosmo0920) [inline fluent-package/debian/fluent-package.lintian-overrides:30]: nits: Shouldn't we use `*` for version numbers?

Comment (kenhys) [reply fluent-package/debian/fluent-package.lintian-overrides]: Thanks, as you mentioned, it should use `*` for it.

File: fluent-package/debian/fluent-package.lintian-overrides [hunk 2]
```diff
@@ -4,17 +4,12 @@ fluent-package: script-not-executable
 fluent-package: python-script-but-no-python-dep
```
Comment (daipom) [inline fluent-package/debian/fluent-package.lintian-overrides:4]: Shouldn't we remove `python-script-but-no-python-dep` too?

```suggestion
```

**Review by daipom — COMMENTED** [review]
Shouldn't we update the man of `td` in this PR?

https://github.com/fluent/fluent-package-builder/blob/7221f3ee05e0458b3962362fa68f40768a615823/fluent-package/debian/td.1

**Review by daipom — APPROVED** [review]
My concerns are the above two points.
Other than those points, this looks good to me!

**Comment by kenhys** [comment]
> Shouldn't we update the man of `td` in this PR?
> 
> https://github.com/fluent/fluent-package-builder/blob/7221f3ee05e0458b3962362fa68f40768a615823/fluent-package/debian/td.1

Good catch!, it should be updated too.

File: fluent-package/debian/fluent-package.lintian-overrides (hunk 2 above)
Comment (kenhys) [reply fluent-package/debian/fluent-package.lintian-overrides]: Thanks, fixed now.

**Comment by kenhys** [comment]
waiting CI again.

**Review by daipom — APPROVED** [review]
LGTM!

**Comment by kenhys** [comment]
Thanks!

**Pull request merged**